### ISSUE TITLE
CMS Decryption fails randomly with BouncyCastle encrypted

### ISSUE THREAD

**Issue by jcarnus** [openssl/openssl#24314]
We are using the Java BouncyCastle lib to create CMS and sign it. We use openssl to decrypt it. This is working pretty well but randomly a CMS cannot be decrypted using OpenSSL but works fine with BouncyCastle.

We tested with various version of BC and OpenSSL and the error is the same .
OpenSSL error is : 
Error decrypting CMS structure
22472:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:crypto\evp\evp_enc.c:612:

We have the CMS and private key to replicate the issue but this cannot be communicated privately for security purpose.

**Comment by t8m** [comment]
What are the algorithms used in the CMS file? This sounds like some encryption key or plaintext padding discrepancy between BC and OpenSSL.

**Comment by t8m** [comment]
Also what openssl version is being used?

**Comment by jcarnus** [comment]
Test are conducted with OpenSSL 1.1.1q  5 Jul 2022

First few char of the first line of the expected file, also obtained with BC or .Net
![image](https://github.com/openssl/openssl/assets/3034768/6171bd1a-5561-439f-86a6-5ee8f3b791ce)

Invalid one with Openssl:
![image](https://github.com/openssl/openssl/assets/3034768/a1d4d296-fd33-4de0-84a5-0e543b275f34)

Also, the invalid file has almost the same size than the good one

**Comment by jcarnus** [comment]
I've just found that setting the -recip in the command line solved the issue. What could cause this issue when not giving the -recip

**Comment by t8m** [comment]
Yes, you must add `-recip` if there are multiple recipients to be able to decrypt messages without such random errors. Due to API hardening the RSA decryption does not return error but actually decrypts to a random key, otherwise Bleichenbacher attacks are possible against the private key due to timing or other side channel leaks.
